File: mobile/src/main/java/packi/day/di/AppComponent.java
package packi.day.di;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import packi.day.WorldApplication;
import packi.day.store.StoreLocator;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        BuildersModule.class})
public interface AppComponent {

    void inject(WorldApplication app);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder storeLocator(StoreLocator locator);

        AppComponent build();
    }
}

File: mainscreen/src/main/java/packi/day/di/AppComponent.kt
package packi.day.di

import dagger.BindsInstance
import dagger.Component
import packi.day.main.FocusCelebrationView
import packi.day.store.StoreLocator
import javax.inject.Singleton

@Singleton
@Component(modules = [FocusCelebrationViewModelProvider::class])
interface AppComponent {

    fun inject(app: FocusCelebrationView)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun storeLocator(locator: StoreLocator): Builder
        fun build(): AppComponent
    }
}